### ISSUE TITLE
Terminate Containers and Clean Workspace between Tests/Playbooks

### ISSUE THREAD

**Issue by billvb** [mgibbs1259/gigantum_testing#25]
**Is your feature request related to a problem? Please describe.**
Running tests leaves a lot of artifacts on the testing machien -- this includes a lot of leftover projects, datasets, docker images/containers, etc. In order to have each test run in a blank slate, the artifacts from prior tests should be cleaned up so as not to interfere.

**Describe the solution you'd like**
After a test:
 * All projects created by a test playbook are cleaned up
 * All Docker images/containers created during the playbook are cleaned up
 * The Gigantum client itself is terminated, and Docker containers are pruned.

**Exit criteria**
Basically, each playbook starts from a blank slate and has to spin up the container itself. 

Note, we need to have some formalization of a "playbook" (as a collection of related tests) and a "test" -- which is a narrow validation of some specific claim. We might not need to restart gigantum between each _test_, but we may need to do so between each _playbook_.

**Comment by billvb** [comment]
Finished per PR
